### PR TITLE
RBD ReclaimSpaceJob and ReclaimSpaceCronJob

### PR DESCRIPTION
**Pull request by keemano**
Test cases covered: -
 1) Reclaiming the space using ReclaimSpaceCronJob
 2) Reclaiming the space using ReclaimSpaceJob when
        -> No space is available to claim
        -> No volume is mounted

### CODE REVIEW

File: tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            # Verify file is deleted
+            try:
+                check_file_existence(pod_obj=pod_obj, file_path=file_path)
+            except CommandFailed as cmdfail:
+                if "No such file or directory" not in str(cmdfail):
+                    raise
+                log.info(f"Verified: File {file_path} deleted.")
+
+            # Create ReclaimSpaceCronJob
+            for type in schedule:
```
Comment (jilju) [inline tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py:111]: The space reclaim will be done in the 'hourly' job. Other scheduled jobs( "midnight", "weekly") will have no space to reclaim.

File: tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                reclaim_space_job = pvc_obj.create_reclaim_space_cronjob(schedule=type)
+
+            # Wait for the Succeeded result of ReclaimSpaceJob
+            try:
+                for reclaim_space_job_yaml in TimeoutSampler(
+                    timeout=120, sleep=5, func=reclaim_space_job.get
+                ):
+                    result = reclaim_space_job_yaml["spec"]["schedule"]
+                    if result == "@" + type:
```
Comment (jilju) [inline tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py:120]: Isn't this verifying only the job having schedule "weekly" which is the  last item in the list 'schedule' ?

File: tests/manage/pv_services/space_reclaim/test_rbd_space_reclaim.py [hunk 3]
```diff
@@ -94,20 +95,166 @@ def test_rbd_space_reclaim(self):
...
+        self.reclaim_space_job(reclaim_space_job)
+
+        # Verify space is reclaimed by checking the used size of the RBD pool
+        used_after_reclaiming_space = fetch_used_size(cbp_name, used_size_after_io)
+        log.info(
+            f"Memory remains intact. Used size after io is {used_after_reclaiming_space}."
+        )
+
+    @polarion_id("OCS-3733")
+    @tier1
```
Comment (jilju) [inline tests/manage/pv_services/space_reclaim/test_rbd_space_reclaim.py:185]:
```suggestion
    @tier2
```

File: tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py (hunk 2 above)
Comment (keemano) [reply tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py:120]: > Isn't this verifying only the job having schedule "weekly" which is the last item in the list 'schedule' ?

This has to be under the for loop. will change it.

**Review by jilju — APPROVED** [review]

**Review by ypersky1980 — APPROVED** [review]
LGTM

File: tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        log.info(f"Cephblock pool name {cbp_name}")
+        used_size_before_io = fetch_used_size(cbp_name)
+        log.info(f"Used size before IO is {used_size_before_io}")
+
+        # Create four 4 GB file
+        for filename in [fio_filename1, fio_filename2, fio_filename3, fio_filename4]:
+            pod_obj.run_io(
+                storage_type="fs",
+                size="4G",
+                runtime=100,
```
Comment (ramkiperiy) [inline tests/manage/pv_services/space_reclaim/test_rbd_reclaimspace_cronjob.py:84]: as Avi mentioned in PR #5936 in given time 4G data will not be written to the PVC.

**Review by ramkiperiy — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**